homeassistant/components/system_bridge/config_flow.py

```diff
@@ -0,0 +1,190 @@
+from homeassistant.const import CONF_API_KEY, CONF_HOST, CONF_PORT
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.helpers.typing import DiscoveryInfoType
+
+from .const import BRIDGE_CONNECTION_ERRORS, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_AUTHENTICATE_DATA_SCHEMA = vol.Schema({vol.Required(CONF_API_KEY): cv.string})
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT, default=9170): cv.string,
+        vol.Required(CONF_API_KEY): cv.string,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data: dict):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    bridge = Bridge(
+        BridgeClient(aiohttp_client.async_get_clientsession(hass)),
+        f"http://{data[CONF_HOST]}:{data[CONF_PORT]}",
+        data[CONF_API_KEY],
+    )
+
+    hostname = data[CONF_HOST]
+    try:
+        async with async_timeout.timeout(30):
+            bridge_os: Os = await bridge.async_get_os()
+            if bridge_os.hostname is not None:
+                hostname = bridge_os.hostname
+            bridge_system: System = await bridge.async_get_system()
+    except BridgeAuthenticationException as exception:
+        _LOGGER.info(exception)
+        raise InvalidAuth from exception
+    except BRIDGE_CONNECTION_ERRORS as exception:
+        _LOGGER.info(exception)
+        raise CannotConnect from exception
+
+    return {"hostname": hostname, "uuid": bridge_system.uuid.os}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for System Bridge."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize flow."""
+        self._name: Optional[str] = None
+        self._input: Optional[Dict[str, Any]] = {}
+
+    async def _async_get_info(self, user_input=None):
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return None, info
+
+        return errors, None
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors, info = await self._async_get_info(user_input)
+        if errors is None:
+            # Check if already configured
+            await self.async_set_unique_id(info["uuid"])
```

Comment:
We don't want to raise in other flows if there's a discovery flow in progress. 
```suggestion
            await self.async_set_unique_id(info["uuid"], raise_on_progress=False)
```